moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useCsvExport.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from "react";
+import { useLazyQuery } from "@apollo/client";
+import { format } from "date-fns";
+import Papa from "papaparse";
+import {
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Grid,
+} from "@mui/material";
+
+export const CsvDownloadDialog = ({ dialogOpen, handleDialogClose }) => (
+  <Dialog
+    open={dialogOpen}
+    onClose={handleDialogClose}
+    aria-labelledby="form-dialog-title"
+  >
+    <DialogTitle id="form-dialog-title"> </DialogTitle>
+    <DialogContent>
+      <Grid container spacing={3}>
+        <Grid item xs={2} lg={2}>
+          <CircularProgress />
+        </Grid>
+        <Grid item xs={10} lg={10}>
+          <DialogContentText>
+            Preparing download, please wait.
+          </DialogContentText>
+        </Grid>
+      </Grid>
+    </DialogContent>
+    <DialogActions>
+      <Button onClick={handleDialogClose} color="primary">
+        Cancel
+      </Button>
```

Comment:
Maybe we should removed this cancel button? It doesn't cancel the CSV download. It only closes the modal.

Reply:
Hmm—yeah this kinda weird. Is it possible have the 'cancel' button actually cancel the download? And I am curious what would/should happen if the download fails for some reason (most likely because the Hasura request bombed or timed out).

Separately, I noticed that if you cancel a download you will not be able to download a file again until you refresh the page.

Curious what you think, Mike—I don't want the cancel to blow up the scope of this work bc i think it's an edge case.

Reply:
I'd have to look at how to the cancel flow works, but it definitely took some effort to catch the cancel button before the export was done. Makes me wonder how often someone would try to do that? Maybe an alert on failure would be good to improve this? I like your idea of handling the error!

Reply:
my vote is to just remove the cancel button. it is hard to catch anyway

Reply:
oh its easy to catch if you have the slowest connection, but if it doesnt stop the download from happening, then we should at least rename the button to say "close"

Reply:
removing it is cool with me 👍